assets/test/Line.test.ts

```diff
@@ -604,19 +607,43 @@ test('can toggle chelsea bridge announcements on and off on Silver Line page', (
     ),
   );
 
-  const chelseaInput = wrapper.find('input[name="chelsea_bridge"]');
-  chelseaInput.simulate('change', { target: { checked: true } });
+  fireEvent.click(screen.getByTestId('chelsea_bridge_toggle'));
   expect(setChelseaBridgeAnnouncements.mock.calls.length).toEqual(1);
   expect(setChelseaBridgeAnnouncements).toHaveBeenCalledWith('auto');
+});
+
+test('can toggle chelsea bridge announcements off on Silver Line page', () => {
```

Comment:
Does this need to be a separate test? It's not always possible, but chaining together multiple interactions can increase the effectiveness of UI tests by potentially catching more complex bugs (with less code, too).

Reply:
Ah I think I might have split this one up because for some reason when using `fireEvent` or `userEvent` to simulate the toggle going off and on, the `setChelseaBridgeAnnouncements` mock was getting called with `auto` both times for some reason. Any thoughts on why that might be happening? I also know that this is more testing implementation details which I understand sort of goes against the principles of RTL. I feel like with this toggle, we may still want to be testing for those details, but if you have other thoughts, I'd be interested in hearing them!

Reply:
Actually scratch that, I think what I described above happens because of how the onChange handler is mocked. The handler is passed down from the top in `ViewerApp` and is supposed to broadcast to all of the phoenix channels first and then rerenders from there.